src/main/java/org/jabref/gui/actions/ActionHelper.java

```diff
@@ -30,10 +31,18 @@ public static BooleanExpression isFieldSetForSelectedEntry(Field field, StateMan
     }
 
     public static BooleanExpression isAnyFieldSetForSelectedEntry(List<Field> fields, StateManager stateManager) {
-        BibEntry entry = stateManager.getSelectedEntries().get(0);
+        ObservableList<BibEntry> selectedEntries = stateManager.getSelectedEntries();
+
+        // binding should be recreated on every right click
+        // not sure why selectedEntries might be empty, see https://github.com/JabRef/jabref/issues/6085
+        if (selectedEntries.isEmpty()) {
+            return Bindings.createBooleanBinding(() -> false, selectedEntries);
+        }
+
+        BibEntry entry = selectedEntries.get(0);
```

Comment:
The binding shouldn't refer to the currently selected entry because this might change in the future and then the binding needs to be reevaluated. The following should work
```suggestion
        ObjectBinding<BibEntry> entry = Bindings.valueAt(selectedEntries, 0);
        return Bindings.createBooleanBinding(() -> {
            if (entry.get() == null)
                return false;
            else
               return entry.get().getFields().stream().anyMatch(fields::contains);
        }, entry, entry.getFieldsObserable());
```